test/BigInt/increment.js

```diff
@@ -0,0 +1,89 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "Increment BigInt literal",
+        body: function () {
+            var x = 123n;
+            assert.isTrue(x == 123n);
+            x++;

```

Comment:
that's is what I did in assign_by_value.

Reply:
Sorry, I don't think I explained very well what I was hoping to see. I thought it might be nice to have a test that uses the result of the increment expression itself, like an assignment or passing it to a method. I see a lot of `x++;` statements on their own, but never a `y = x++` or a `f(x++)`.

---
In reply to: [226483957](https://github.com/Microsoft/ChakraCore/pull/5788#discussion_r226483957) [](ancestors = 226483957)

Reply:
Got you but I don't think this PR addresses assign with pre/post inc/dec. This is in my TODO lists

Reply:
I expect that if you start AbsoluteIncrement and AbsoluteDecrement with cloning the existing BigInt, then everything else should work out correctly.

---
In reply to: [226508075](https://github.com/Microsoft/ChakraCore/pull/5788#discussion_r226508075) [](ancestors = 226508075)

Reply:
@sethbrenith I think it is more to that (tried, didn't work). Also, when we do `y=x++`, the existing flow do convert x to number which can not apply directly to bigint.